trino-connector/src/main/java/org/apache/gravitino/trino/connector/catalog/CatalogRegister.java

```diff
@@ -171,33 +171,34 @@ public void registerCatalog(String name, GravitinoCatalog catalog) {
   }
 
   private boolean checkCatalogExist(String name) {
-    String showCatalogCommand = String.format("SHOW CATALOGS like '%s'", name);
-    Exception failedException = null;
-    try {
-      int retries = EXECUTE_QUERY_MAX_RETRIES;
-      while (retries-- > 0) {
-        try (Statement statement = connection.createStatement()) {
-          // check the catalog is already created
-          statement.execute(showCatalogCommand);
-          ResultSet rs = statement.getResultSet();
-          while (rs.next()) {
-            String catalogName = rs.getString(1);
-            if (catalogName.equals(name) || catalogName.equals("\"" + name + "\"")) {
-              return true;
-            }
+    String showCatalogCommand = String.format("SHOW CATALOGS LIKE '%s'", name);
+    long endTime = System.currentTimeMillis() + EXECUTE_QUERY_TIMEOUT_MS;
+
+    do {
+      try (Statement statement = connection.createStatement();
+           ResultSet rs = statement.executeQuery(showCatalogCommand)) {
+
+        while (rs.next()) {
+          if (name.equalsIgnoreCase(rs.getString(1).replace("\"", ""))) {
+            return true;
           }
-          return false;
-        } catch (Exception e) {
-          failedException = e;
-          LOG.warn("Execute command failed: {}, ", showCatalogCommand, e);
-          Thread.sleep(EXECUTE_QUERY_BACKOFF_TIME_SECOND * 1000);
+        }
+        return false;
+      } catch (SQLException e) {
```

Comment:
Why do you only handle the exception type of SQLException

Reply:
Actually it should be **Exception e ** insteadof **SQLException**. Shall I make this change?

Reply:
I'm not sure, you can add some code to handle other exception types

Reply:
sure, i'll add checks for other exception as well. Also, am i going correct direction so far?

Reply:
That's ok. 
Here, it is important to exception handling and retry after an exception and the retry backoff time. If you can handle exceptions more precisely by type, that would be better, as some exceptions will still fail even after retries